debug/p2p/p2pDebugger.go

```diff
@@ -0,0 +1,213 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/display"
+)
+
+var log = logger.GetOrCreate("debug/p2p")
+
+const printTimeOneSecond = time.Second //if this needs to be changed, remember to divide the values when computing metrics
+
+type metric struct {
+	topic string
+
+	incomingSize         uint64
+	incomingNum          uint32
+	incomingRejectedSize uint64
+	incomingRejectedNum  uint32
+
+	outgoingSize         uint64
+	outgoingNum          uint32
+	outgoingRejectedSize uint64
+	outgoingRejectedNum  uint32
+}
+
+func (m *metric) stringify() []string {
+	return []string{
+		m.topic,
+		fmt.Sprintf("%d / %s/s", m.incomingNum, core.ConvertBytes(m.incomingSize)),
+		fmt.Sprintf("%d / %s/s", m.incomingRejectedNum, core.ConvertBytes(m.incomingRejectedSize)),
+		fmt.Sprintf("%d / %s/s", m.outgoingNum, core.ConvertBytes(m.outgoingSize)),
+		fmt.Sprintf("%d / %s/s", m.outgoingRejectedNum, core.ConvertBytes(m.outgoingRejectedSize)),
+	}
+}
+
+type p2pDebugger struct {
+	selfPeerId          core.PeerID
+	mut                 sync.Mutex
+	data                map[string]*metric
+	cancelFunc          func()
+	shouldProcessDataFn func() bool
```

Comment:
no need for these 2 function pointers - shouldProcessDataFn and printStringFn

Reply:
I need them to properly test this struct